service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyServiceHandler.java

```diff
@@ -94,7 +108,15 @@ public void initializeSession(IoSession session, ConnectFuture future) {
             HttpConnectSession connectSession = (HttpConnectSession) session;
             connectSession.setVersion(acceptSession.getVersion());
             connectSession.setMethod(acceptSession.getMethod());
-            connectSession.setRequestURI(acceptSession.getRequestURI());
+            String acceptPathInfo = acceptSession.getPathInfo().toString();
+            if (acceptPathInfo.startsWith("/")) {
```

Comment:
Looking in the rfc2396 ( at section 5.2 steps 5 and 6 ) for resolving a base URI against a reference URI, I see that if the acceptPath begins with "/", then after .resolve() the whole path in connect is removed. 
If only query is appended to the accept in config in the request, then acceptPathInfo will start with "?", that is why I used this validation.